### PR TITLE
feat(#557): added redirect modal to e.afetharita.com for slow connections

### LINKED ISSUE #557: feat: Redirect to e.afetharita.com for areas with low connectivity and mobile devices
## Define your request
Redirect to e.afetharita.com for areas with low connectivity and mobile devices. It loads faster and we can get details more quickly.

** discord username: Denizsl#2537 **

## Define the alternatives you have in mind
--

## Additional context
Current performance is not enough. Getting the data rapidly is very important in times like these.

### PR DESCRIPTION
**Pull request by denizsult**
# Description
It shows a modal if the user's internet speed is low. By default, this modal is active if the request is delayed by 6 seconds. It offers the person an option for e.afetharita.com.

**discord username: Denizsl#2537**

**closes #557**

Please describe your changes. Also describe your aim and content. Do not forget to list the dependencies required caused by those changes.
Screen Shot 2023-02-11 at 14 29 57
Screen Shot 2023-02-11 at 14 30 12

## Things to check before creating a PR
- [x] I have inspected my topic, checked.
- [x] If it is a core feature, executed detailed tests.

## Creating PR rules
- [] PR must be created for an issue with approved tag. Otherwise PR will be rejected.
- [] Relevant issue number: The issue number related to PR must be attached to head of PR header, after prefix after prefix # must be attached in parenthesis. A header like this could be used "prefix(#issue_number): PR header" .
- [] A descriptive and understandable title: The PR title should clearly describe the nature and purpose of the changes. The PR title should be the first thing displayed when the PR is opened. And it should follow the semantic commit rules. For example, a title like "docs(#issueId): Add README.md" can be used.
- [] Related file selection: Only relevant files should be touched and no other files should be affected.
- [] Format and Lint Suitability : The code should be made in accordance with a certain format standard and examined according to the lint rules.
- [] Clean commit history: The commits where the changes are made should be clear and organized.
- [] Opening PR when job is completed: PR should be opened when job is completed and sent for review by other team members.

## Changes

- [x] A new feature (a change that adds a new feature, not a breaking change)
- [ ] A new refactor (a change that is not a breaking change, that improves the readability or performance of the code)
- [ ] A breaking change
- [ ] Documentation change

# How were these changes tested?

Please describe the tests you did to test the changes you made. Please also specify your test configuration.
I tested adding a promise delay above the swr fetch.

**Test Configuration**:

* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

**Comment by eraygundogmus** [comment]
@fdemir we are gonna add some features too. i announced on discord. could you update the map?

File: pages/index.tsx [hunk 1]
```diff
@@ -106,7 +109,11 @@ export default function Home({ deviceType, singleItemDetail }: Props) {
     dataFetcher,
     {
       isPaused: () => !coordinatesAndEventType,
-      onLoadingSlow: () => setSlowLoading(true),
+      onLoadingSlow: () => {
```
Comment (cdagli) [inline pages/index.tsx:112]: @denizsult I understood the intention but I think here is a very late stage to show this modal. At this stage, user already loads the whole application and makes the request next. If they reach to that point without any issues I'd say they already have a device&connection that can make afetharita.com work. 

I believe we need something more like this https://stackoverflow.com/a/9011781 

What do you think?

**Comment by denizsult** [comment]
> @fdemir we are gonna add some features too. i announced on discord. could you update the map?

I missed the Discord announcement.  How can I reach the changes about the map? The Forked repo is up to date.

File: pages/index.tsx (hunk 1 above)
Comment (denizsult) [reply pages/index.tsx:112]: What he said may be true. Let me research a little more. Thanks a lot!

Comment (cdagli) [reply pages/index.tsx]: @denizsult I am adding WIP label then, thank you!

Comment (denizsult) [reply pages/index.tsx]: @cdagli  I implemented the solution we talked about.  Also, it should not be taken without the knowledge of the @fdemir.

**Comment by usirin** [comment]
Closing this as the original issue is closed, we can reopen later when we decide if we want to move forward with this

**Pull request closed without merging**